src/coupling/hm_iterative.cc

```diff
@@ -251,11 +264,29 @@ void HM_Iterative::update_potential()
             double potential = -alpha*density*gravity*pressure;
         
             potential_vec_.set( dof_indices[side.side_idx()], potential );
+
+            // The reference potential is applied only on dirichlet and total_flux b.c.,
+            // i.e. where only mechanical traction is prescribed.
+            if (side.side().is_boundary() &&
+                    (flow_->data().bc_type.value(side.centre(), side.cond().element_accessor()) == DarcyMH::EqData::dirichlet ||
```

Comment:
We can not use .value to evaluate fields anymore. Need to refactor this as an auxiliary assembly class.